sdk/opendp/smartnoise/evaluation/learner/_bandit.py

```diff
@@ -0,0 +1,75 @@
+import os
+import numpy as np
+import pandas as pd 
+
+from opendp.smartnoise.evaluation.params._learner_params import LearnerParams
+from opendp.smartnoise.evaluation.learner._generate import Grammar
+from opendp.smartnoise.evaluation.learner.util import create_simulated_dataset, generate_neighbors, write_to_csv
+
+from opendp.smartnoise.evaluation.params._learner_params import LearnerParams
+from opendp.smartnoise.evaluation.params._privacy_params import PrivacyParams
+from opendp.smartnoise.evaluation.params._eval_params import EvaluatorParams
+from opendp.smartnoise.evaluation.params._dataset_params import DatasetParams
+from opendp.smartnoise.evaluation.evaluator._dp_evaluator import DPEvaluator
+from opendp.smartnoise.sql import PandasReader
+from dp_singleton_query import DPSingletonQuery
+
+
+class bandit():
+    def __init__(self):
+        self.pp = PrivacyParams(epsilon=1.0)
+        self.ev = EvaluatorParams(repeat_count=100)
+        self.dd = DatasetParams(dataset_size=500)
+        self.pa = DPSingletonQuery()
+ 
+    def generate_query(self, ep: LearnerParams):
+        #generate query pool
+        select_path = os.path.join(os.path.dirname(__file__),"select.cfg")
+        print(select_path)
+        with open (select_path, "r") as cfg:
+            rules=cfg.readlines()
+            grammar = Grammar(ep)
+            numofquery = ep.numofquery
+            grammar.load(rules)
+
+
+        text_file = open("querypool.txt", "w")
```

Comment:
with open("querypool.txt", "w") as test_file:

Reply:
the code below  and before the close should be tabbed in to be within the with block